### PR TITLE
conformance: fetch Gateway in GatewayMustHaveLatestConditions loop

### PR DESCRIPTION
**Pull request by skriss**
**What type of PR is this?**
/kind flake
/area conformance

**What this PR does / why we need it**:
Fetches the Gateway from the client each time through
the polling loop in GatewayMustHaveLatestConditions,
so updates to the conditions can be observed.

The current code is checking the same object over and
over again.

**Which issue(s) this PR fixes**:
Updates #1883

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/tests/gateway-modify-listeners.go
Comment (sunjayBhatia) [inline conformance/tests/gateway-modify-listeners.go:58]: should we move this after the namespace ready check but before fetching the "original" gateway? (similar for the other test in this file)

Comment (skriss) [reply conformance/tests/gateway-modify-listeners.go:58]: It looks like `original` is only being used for spec fields, so shouldn't matter too much one way or another, but I'm fine changing the order for readability.

Comment (sunjayBhatia) [reply conformance/tests/gateway-modify-listeners.go:58]: yeah good point, the original Generation field shouldnt change since the resource shouldnt be updated from when we create it, so I'll defer to you if you think it makes the code more readable to fetch the Gateway after checking the status

**Review by sunjayBhatia — APPROVED** [review]
LGTM passing with Contour, but since Steve and I both are testing against the same implementation, should get a review from someone else as well

File: conformance/tests/gateway-modify-listeners.go
Comment (skriss) [reply conformance/tests/gateway-modify-listeners.go:55]: https://github.com/kubernetes-sigs/gateway-api/pull/2203/commits/991056d97d2d192866d08ee5dfa9e66abda2c047

**Comment by robscott** [comment]
Thanks @skriss!

/lgtm
/approve

**Pull request merged**
